azure/services/async/async_test.go

```diff
@@ -227,7 +230,7 @@ func TestServiceDeleteResource(t *testing.T) {
 					s.GetLongRunningOperationState(resourceName, serviceName, infrav1.DeleteFuture).Return(validDeleteFuture),
 					d.DeleteAsync(gomockinternal.AContext(), gomock.AssignableToTypeOf(azureResourceGetterType), gomock.Any()).Return(fakePoller[MockDeleter](g, http.StatusAccepted), context.DeadlineExceeded),
 					s.SetLongRunningOperationState(gomock.AssignableToTypeOf(&infrav1.Future{})),
-				)
+					s.DefaultedReconcilerRequeue().Return(reconciler.DefaultReconcilerRequeue))
```

Comment:
```suggestion
					s.DefaultedReconcilerRequeue().Return(reconciler.DefaultReconcilerRequeue),
				)
```
Nitpick for consistency with the other test cases.